feat(android): Add custom masking options for session replay (#11624)
getsentry/sentry-docs: docs/platforms/android/session-replay/index.mdx

## Privacy

The SDK is recording and aggressively masking all text, images and webviews. Please don’t turn it off if you have sensitive data in your app.
However, if you're working on a mobile app that's free of PII or other types of private data, you can opt out of the default text and image masking settings. To learn more about Session Replay privacy, [read our docs](/platforms/android/session-replay/privacy/).

<Note>

> If you encounter any other data not being masked with the default settings, please let us know through a [GitHub issue](https://github.com/getsentry/sentry-java/issues/new?assignees=&labels=Platform%3A+Android%2CType%3A+Bug&projects=&template=bug_report_android.yml).

Reviewer: ```suggestion
If you find that any other data isn't being masked with the default settings, please let us know by creating a [GitHub issue](https://github.com/getsentry/sentry-java/issues/new?assignees=&labels=Platform%3A+Android%2CType%3A+Bug&projects=&template=bug_report_android.yml).
```

</Note>

To disable masking altogether (not to be used on applications with sensitive data):

```kotlin

> options.experimental.sessionReplay.maskAllText = false

Author: yeah, good idea, I'll add it back 👍

options.experimental.sessionReplay.maskAllImages = false

> ```

Author: ```suggestion
// if you're on version < 7.15.0
options.experimental.sessionReplay.redactAllText = false
options.experimental.sessionReplay.redactAllImages = false
```
```

## FAQ

> Q: Why are parts of my replay not masked?
> A: Text fields, input fields, images, video players and webviews are all masked by default. Local assets, such as colors or vector drawables, are not masked due to their low likelihood of containing PII. If you encounter a view/component that should be masked by default, consider opening a [GitHub issue](https://github.com/getsentry/sentry-java/issues).
> 

Reviewer: ```suggestion
Q: Why are parts of my replay not masked?
A: Text fields, input fields, images, video players and webviews are all masked by default. Local assets, such as colors or vector drawables, aren't masked because the likelihood of these assets containing PII is low. If you encounter a view/component that should be masked by default, consider opening a [GitHub issue](https://github.com/getsentry/sentry-java/issues).

```

Q: Does Session Replay work with Jetpack Compose?

> A: Yes, by default text, input field and image composables should be masked. Masking within embedded android views (`AndroidView`)
> in Compose is currently not supported. If you encounter composables that are not masked but should be, consider opening a [GitHub issue](https://github.com/getsentry/sentry-java/issues).

Reviewer: ```suggestion
A: Yes, by default, text, input field, and image composables should be masked. Masking within embedded Android views (`AndroidView`)
in Compose isn't currently supported. If you encounter composables that aren't masked but should be, consider opening a [GitHub issue](https://github.com/getsentry/sentry-java/issues).
```
